### ISSUE TITLE
Control when our pipeline extracts data updates

### ISSUE THREAD

**Issue by vingkan** [scarletstudio/transithealth#5]
Some data systems are designed to ingest data in real-time, but for this project, we want to control when we get the latest data. The main reason is because our (compressed) database is part of our codebase, which is usually not the case for software projects. Some datasets we will add to TransitHealth are updated every few weeks or even days. If we did not have a filter to limit the latest data by date, then our pull requests would have lots of extra diffs for the compressed database files and updated database tests.

When @Andrew-Tolentino tried to merge his PR to add income data, the database tests failed unexpectedly for the `covid_spread` table, even though he did not change that table.

![188186239_900313170530080_4489177090016078216_n](https://user-images.githubusercontent.com/11896652/119208073-34c9c100-ba66-11eb-9025-588d430f85b8.png)

The reason was that when he ran `make clean && make` and the extract step for the COVID cases dataset from the City data portal fetched the latest updates. That dataset gets updated roughly every week.

To pass the tests and merge, he [had to update the database tests](https://github.com/scarletstudio/transithealth/commit/e6f420ead9eec4f2b6148378e35b905b99e62a63).

This might be confusing/surprising if someone runs the pipeline from scratch while working on another change, so maybe we want to do something to control when we ingest the latest data. For example, the rideshare trips dataset is updated every three months, so we are expecting to get a big update, possibly during the internship.

One idea could be to add an end date to the extract script so that we do not automatically fetch new data. Then we can fetch the updates on our own schedule, sending a specific PR to increase the end date, run the pipeline from scratch, and update the tests.

**Comment by vingkan** [comment]
Another important requirement that I did not think of originally: we want to limit end date based on the data, not based on when the update happened.

Since I last ran the offline pipeline on May 23rd, I tried to set May 23rd as the limit for the COVID cases dataset, but when I reran the pipeline, I still got more rows. The reason is that when I extracted the COVID cases dataset on May 23rd, the latest data was for May 8th. Then on May 27nd, the City uploaded data for the weeks ending in May 15th and May 22nd.

So when we update the filter in the extract query to get the latest data, instead of changing it to the date of the run, we should check the data portal to find out the date of the latest data (from the table, not from the update metadata) and use that as the filter end date.
